File: javascript/easy/hammingDistance.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
    let xor = x ^ y, distance = 0;
    while(xor > 0) {
        if((xor & 1) == 1) {
            distance++;
        }
        xor >>= 1;
    }
    return distance;
};

File: ruby/easy/hamming_distance.rb
# @param {Integer} x
# @param {Integer} y
# @return {Integer}
def hamming_distance(x, y)
    xor = x ^ y
    hamming_distance = 0
    while xor > 0
        hamming_distance += 1 if (xor & 1) == 1
        xor >>= 1
    end
    hamming_distance
end
